portal-impl/src/com/liferay/portal/util/PropsValues.java

```diff
@@ -831,21 +827,9 @@ public class PropsValues {
 			PropsUtil.get(
 				PropsKeys.FACEBOOK_CONNECT_VERIFIED_ACCOUNT_REQUIRED));
 
-	public static final JSONObject FEATURE_FLAGS_JSON_OBJECT =
-		new JSONObjectImpl() {
-			{
-				Properties properties = PropsUtil.getProperties(
-					"feature.flag.", true);
-
-				for (Map.Entry<Object, Object> property :
-						properties.entrySet()) {
-
-					put(
-						GetterUtil.getString(property.getKey()),
-						GetterUtil.getBoolean(property.getValue()));
```

Comment:
Hi, @dantewang. This was necessary so we can make frontend receive **booleans** instead of **strings**.
It makes FE engineers' lives easier if they just need to check `if (!Liferay.FeatureFlags['LPS-XXXXX'])` instead of `if (!Liferay.FeatureFlags['LPS-XXXXX'] || Liferay.FeatureFlags['LPS-XXXXX'] === 'false')`. In other words, `"false" != false`.

Reply:
@rodrigopaulino My target is to avoid JSON operation at hot path, so I think I can keep your logic but still use the string (`JSONObject.toString()`) as the constant?

Reply:
Feel free to adjust as you think it's best, as long as the the code keeps mapping **strings** to **booleans**.

Reply:
Hey, @dantewang. Since you're transforming the json to string, you may need to parse this to an object in javascript. Also, I don't think the JSON_OBJECT suffix is accurate to be naming that constant anymore.